scripts/authors-by-version.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+
+# Example usage:
+# mkdir -p reference
+# for VERSION in v1.0.0 HEAD; do
+#   ./scripts/authors-by-version.sh $VERSION > reference/authors-$VERSION.md
+# done
+
+# Authors, comitters, co-authors
+get_authors() {
+    local git_tag="$1"
+    shift  # Shift arguments to get the files
+    local files="$@"
+    git --no-pager log "$git_tag" -- $files | \
+        sed -n -e 's/^Author: //p' -e 's/^Committer: //p' -e 's/^.*Co-authored-by: //p' | \
+        sort | uniq | sed -e 's/^/- /'
+}
+
+# Ensure tag argument is provided
+if [ "$#" -lt 1 ]; then
+    echo "Usage: $0 <git-tag>"
+    exit 1
+fi
+
+GIT_TAG="$1"
+
+if [ ! $(git tag -l "$GIT_TAG") ] && [ ! "$GIT_TAG" == "HEAD" ]; then
+    echo "<git-tag> '$GIT_TAG' does not exist"
+    exit 1
+fi
+
+# Check if translations
+git show "$GIT_TAG":i18n > /dev/null 2>&1 && i18n=true || i18n=false
+
+
+# Output
+cat <<EOF
+# $GIT_TAG Authors
+
+⚠️ Do not edit this file directly.
+It is [auto-generated](../scripts/authors-by-version.sh) from Git history of source files in this repo.
+
+ℹ️ Note this is purely a convenience for referencing in academic papers.
+See the full Git history of specific files for additional context.
+
+EOF
+
+# If translations exist, output a note about language and translations
+if [ $i18n = true ]; then
+    cat <<EOF
+## Language and Translations
+
+The GitOps Principles and Glossary are [authored](#authors-list) in English.
+See farther below for a list of [translators](#translators-list) by language.
+
+EOF
```

Comment:
You can use `<<-EOF` for formatting

```shell
if [ $i18n = true ]; then
	cat <<-EOF
	## Language and Translations
	The GitOps Principles and Glossary are [authored](#authors-list) in English.
	See farther below for a list of [translators](#translators-list) by language.
	EOF
fi
```

I haven't tested it with spaces (I am a TAB person)

Reply:
Yeah, sadly doesn't work with spaces

Reply:
![image](https://y.yarn.co/14c6c794-24db-451a-924c-570ab0705908_text.gif)